lib/trino-hdfs/src/test/java/io/trino/filesystem/hdfs/TestHdfsFileSystemLocal.java

```diff
@@ -0,0 +1,133 @@
+import io.trino.hdfs.HdfsConfigurationInitializer;
+import io.trino.hdfs.HdfsContext;
+import io.trino.hdfs.HdfsEnvironment;
+import io.trino.hdfs.TrinoHdfsFileSystemStats;
+import io.trino.hdfs.authentication.NoHdfsAuthentication;
+import io.trino.spi.security.ConnectorIdentity;
+import org.apache.hadoop.fs.RawLocalFileSystem;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.util.Comparator.reverseOrder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestHdfsFileSystemLocal
+        extends AbstractTestTrinoFileSystem
+{
+    private TrinoFileSystem fileSystem;
+    private Path tempDirectory;
+
+    @BeforeAll
+    void beforeAll()
+            throws IOException
+    {
+        RawLocalFileSystem.useStatIfAvailable();
+        DynamicConfigurationProvider viewFs = (config, context, uri) ->
+                config.set("fs.viewfs.mounttable.abc.linkFallback", tempDirectory.toAbsolutePath().toUri().toString());
+
+        HdfsConfig hdfsConfig = new HdfsConfig();
+        HdfsConfiguration hdfsConfiguration = new DynamicHdfsConfiguration(new HdfsConfigurationInitializer(hdfsConfig), Set.of(viewFs));
+        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(hdfsConfiguration, hdfsConfig, new NoHdfsAuthentication());
+        HdfsContext hdfsContext = new HdfsContext(ConnectorIdentity.ofUser("test"));
+        TrinoHdfsFileSystemStats stats = new TrinoHdfsFileSystemStats();
+
+        tempDirectory = Files.createTempDirectory("test");
+        fileSystem = new HdfsFileSystem(hdfsEnvironment, hdfsContext, stats);
+    }
+
+    @AfterEach
+    void afterEach()
+            throws IOException
+    {
+        cleanupFiles();
+    }
+
+    @AfterAll
+    void afterAll()
+            throws IOException
+    {
+        Files.delete(tempDirectory);
+    }
+
+    private void cleanupFiles()
+            throws IOException
+    {
+        // tests will leave directories
+        try (Stream<Path> walk = Files.walk(tempDirectory)) {
+            Iterator<Path> iterator = walk.sorted(reverseOrder()).iterator();
+            while (iterator.hasNext()) {
+                Path path = iterator.next();
+                if (!path.equals(tempDirectory)) {
+                    Files.delete(path);
+                }
+            }
+        }
+    }
+
+    @Override
+    protected boolean isHierarchical()
+    {
+        return true;
+    }
+
+    @Override
+    protected TrinoFileSystem getFileSystem()
+    {
+        return fileSystem;
+    }
+
+    @Override
+    protected Location getRootLocation()
+    {
+        return Location.of("viewfs://abc/");
```

Comment:
Should we name this test `TestHdfsFileSystemViewFs`?

Reply:
This is testing the local file system, using ViewFS to create a sandbox. The important behavior is that the operations are happening against the local file system. ViewFS shouldn't affect that.